src/metal/rt64_metal.cpp

```diff
@@ -2632,32 +2672,48 @@ namespace RT64 {
         if (activeRenderEncoder == nullptr) {
             NS::AutoreleasePool *releasePool = NS::AutoreleasePool::alloc()->init();
 
-            // target frame buffer & sample positions affect the descriptor
             MTL::RenderPassDescriptor *renderDescriptor = MTL::RenderPassDescriptor::renderPassDescriptor();
 
+            // Setup color attachments
             for (uint32_t i = 0; i < targetFramebuffer->colorAttachments.size(); i++) {
                 MTL::RenderPassColorAttachmentDescriptor *colorAttachment = renderDescriptor->colorAttachments()->object(i);
                 colorAttachment->setTexture(targetFramebuffer->colorAttachments[i]->getTexture());
-                colorAttachment->setLoadAction(MTL::LoadActionLoad);
+                colorAttachment->setLoadAction(pendingClears.active ? pendingClears.initialAction[i] : MTL::LoadActionLoad);
+                if (pendingClears.active && pendingClears.initialAction[i] == MTL::LoadActionClear) {
+                    colorAttachment->setClearColor(mapClearColor(pendingClears.clearValues[i].color));
+                }
```

Comment:
It should be possible to skip the check for 'active' here. Both initial action and clear colors should already be in the right value. Presumably, setting a clear color won't do anything either if the action isn't clear as well.